File: src/main/java/logan/pickpocket/user/PlayerConfiguration.java
package logan.pickpocket.user;

import logan.api.config.YamlConfigurationUtil;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

/**
 * Created by Tre on 1/18/2016.
 */
public class PlayerConfiguration {

    private File file;
    private YamlConfiguration configuration;

    public PlayerConfiguration(String directory, String fileName) {
        this.file = new File(directory, fileName);
        this.configuration = YamlConfiguration.loadConfiguration(file);

        // We set these incase the users configuration doesn't contain
        // one of these keys (due to an error, or upgrading of the plugin where
        // the configuration file already exists causing the absence of new keys).
        YamlConfigurationUtil.setIfNotSet(configuration, "admin", false);
        YamlConfigurationUtil.setIfNotSet(configuration, "bypass", false);
        YamlConfigurationUtil.setIfNotSet(configuration, "exempt", false);
        YamlConfigurationUtil.setIfNotSet(configuration, "steals", 0);
        YamlConfigurationUtil.setIfNotSet(configuration, "thiefProfile", "default");
        try {
            configuration.save(file);
        } catch (IOException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
    }

    public YamlConfiguration getConfiguration() {
        return configuration;
    }

    public void setConfiguration(YamlConfiguration configuration) {
        this.configuration = configuration;
    }

    public boolean isAdmin() {
        return configuration.getBoolean("admin");
    }

    public void setAdmin(boolean value) {
        configuration.set("admin", value);
    }

    public boolean isBypassing() {
        return configuration.getBoolean("bypass");
    }

    public void setBypassing(boolean value) {
        configuration.set("bypass", value);
    }

    public boolean isExempt() {
        return configuration.getBoolean("exempt");
    }

    public void setExempt(boolean value) {
        configuration.set("exempt", value);
    }

    public int getStealCount() {
        return configuration.getInt("steals");
    }

    public void setStealCount(int value) {
        configuration.set("steals", value);
    }

    public String getThiefProfile() {
        return configuration.getString("thiefProfile");
    }

    public void setThiefProfile(String value) {
        configuration.set("thiefProfile", value);
    }
}

File: src/logan/pickpocket/user/PlayerConfiguration.kt
package logan.pickpocket.user

import logan.api.config.BasicConfiguration
import logan.api.config.setIfNotSet
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

/**
 * Created by Tre on 1/18/2016.
 */
class PlayerConfiguration(directory: String, fileName: String) : BasicConfiguration {

    override var file: File = File(directory, fileName)
    override var configuration: YamlConfiguration = YamlConfiguration.loadConfiguration(file)

    var isAdmin
        get() = configuration.getBoolean("admin")
        set(value) = configuration.set("admin", value)
    var isBypassing
        get() = configuration.getBoolean("bypass")
        set(value) = configuration.set("bypass", value)
    var isExempt
        get() = configuration.getBoolean("exempt")
        set(value) { configuration.set("exempt", value) }
    var stealCount
        get() = configuration.getInt("steals")
        set(value) = configuration.set("steals", value)
    var isParticipating
        get() = configuration.getBoolean("participating")
        set(value) = configuration.set("participating", value)
    var thiefProfile
        get() = configuration.getString("thiefProfile")
        set(value) { configuration.set("thiefProfile", value) }

    init {
        configuration.setIfNotSet("admin", false)
        configuration.setIfNotSet("bypass", false)
        configuration.setIfNotSet("exempt", false)
        configuration.setIfNotSet("steals", 0)
        configuration.setIfNotSet("participating", false)
        configuration.setIfNotSet("thiefProfile", "default")
        save()
    }
}